sqlglot/optimizer/pushdown_projections.py

```diff
@@ -112,6 +115,9 @@ def _remove_unused_selections(scope, parent_selections, schema, alias_count):
                 star = True
             removed = True
 
+        if not is_agg and selection.find(exp.AggFunc):
```

Comment:
Will this also be the case in:
```
SELECT
  a
FROM x
GROUP BY a
```
?

Reply:
no only if there's an agg func. added an additional test